pennylane/templates/subroutines/qsvt.py

```diff
@@ -507,6 +482,93 @@ class QSVT(Operation):
+        compatible.
+
+        The following example applies the polynomial :math:`p(x) = -x + 0.5x^3 + 0.5x^5` to an
+        arbitrary hermitian matrix using :class:`~.BlockEncode` for block encoding. Note that we
+        can simply use :class:`~.FABLE` for block encoding as well.
+
+        .. code-block::
+
+            poly = [0, -1, 0, 0.5, 0, 0.5]
+            angles = qml.poly_to_angles(poly, "QSVT")
+            input_matrix = np.array([[0.2, 0.1], [0.1, -0.1]])
+
+            wires = [1, 2]
+            block_encode = qml.BlockEncode(input_matrix, wires=wires)
+            projectors = [
+                qml.PCPhase(angles[i], dim=len(input_matrix), wires=wires)
+                for i in range(len(angles))
+            ]
+
+            dev = qml.device("default.qubit")
+            @qml.qnode(dev)
+            def circuit():
+                qml.QSVT(block_encode, projectors)
+                return qml.state()
+
+            matrix = qml.matrix(circuit, wire_order=[0, 1,2])()
+
+        .. code-block:: pycon
+
+            >>> print(np.round(matrix[: len(input_matrix), : len(input_matrix)], 4).real)
+            [[-0.1942 -0.0979]
+             [-0.0979  0.0995]]
+
+        If the initial operator for which we apply the singular value transformation is a linear
+        combination of unitaries, e.g., a Hamiltonian, it can be blcok-encoded with operations
+        such as :class:`~.PrepSelPrep` or :class:`~.Qubitization`. Note that both of these oprations
+        have a proper gate decomposition. The following example applies the polynomial
+        :math:`p(x) = -x + 0.5x^3 + 0.5x^5` to the Hamiltonian :math:`H = 0.1X_3 - 0.7X_3Z_4 - 0.2Z_3Y_4`,
+        In this example it will be applied the polynomial :math:`p(x) = -x + 0.5x^3 + 0.5x^5` to
+        blcok-encoded with :class:`~.PrepSelPrep`.
+
+        .. code-block::
+
+            poly = np.array([0, -1, 0, 0.5, 0, 0.5])
+            H = 0.1 * qml.X(3) - 0.7 * qml.X(3) @ qml.Z(4) - 0.2 * qml.Z(3)
+
+            control_wires = [1, 2]
+            block_encode = qml.PrepSelPrep(H, control=control_wires)
+            angles = qml.poly_to_angles(poly, "QSVT")
+
+            projectors = [
+                qml.PCPhase(angles[i], dim=2 ** len(H.wires), wires=control_wires + H.wires)
+                for i in range(len(angles))
+            ]
+
+            dev = qml.device("default.qubit")
+
+            @qml.qnode(dev)
+            def circuit():
+                qml.QSVT(block_encode, projectors)
+                return qml.state()
+
+            matrix = qml.matrix(circuit, wire_order=[0] + control_wires + H.wires)()[: 2 ** len(H.wires), : 2 ** len(H.wires)]
+
+        .. code-block:: pycon
+
+            >>> print(np.round(matrix, 4).real)
+            [[ 0.144   0.      0.432   0.    ]
+             [ 0.      0.0858  0.     -0.343 ]
+             [ 0.432   0.     -0.144   0.    ]
+             [ 0.     -0.343   0.     -0.0858]]
```

Comment:
```suggestion
        .. code-block:: pycon

            >>> print(np.round(matrix, 4).real)
            [[ 0.144   0.      0.432   0.    ]
             [ 0.      0.0858  0.     -0.343 ]
             [ 0.432   0.     -0.144   0.    ]
             [ 0.     -0.343   0.     -0.0858]]
        
        Note that we are taking the real part of the matrix by accessing its values. The hardware compatible way to do this would be by adding the operator QSVT and its adjoint as shown in `this demo <https://pennylane.ai/qml/demos/tutorial_apply_qsvt>`.      
        
```

What do you think about writing something like this?

Reply:
Better to update the examples and just run the circuit.